mission/mission_planner/mission_planner/mission_planner.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/env python3
+
+import sys
+import rclpy
+from rclpy.node import Node
+from vortex_msgs.srv import MissionPlanner
+
+class MissionPlannerClient(Node):
+    """
+    A ROS2 client node for interacting with the MissionPlanner service.
+    """
+    def __init__(self):
+        """
+        Initializes the client node, creates a client for the MissionPlanner service,
+        and waits for the service to become available.
+        """
+        super().__init__('mission_planner_client')
+        self.client = self.create_client(MissionPlanner, 'mission_planner')
+        while not self.client.wait_for_service(timeout_sec=1.0):
+            self.get_logger().info('Service not available, waiting again...')
+        self.req = MissionPlanner.Request()
+
+    def send_request(self, ox: list, oy: list, sx: int, sy: int, gx: int, gy: int):
+        """
+        Sends an asynchronous request to the MissionPlanner service with obstacle locations,
+        start, and goal positions.
+
+        Args:
+            ox (list): The x-coordinates of obstacles.
+            oy (list): The y-coordinates of obstacles.
+            sx (int): The x-coordinate of the start position.
+            sy (int): The y-coordinate of the start position.
+            gx (int): The x-coordinate of the goal position.
+            gy (int): The y-coordinate of the goal position.
+        """
+        self.req.ox = ox
+        self.req.oy = oy
+        self.req.sx = sx
+        self.req.sy = sy
+        self.req.gx = gx
+        self.req.gy = gy
+        self.future = self.client.call_async(self.req)
+        self.get_logger().info('MissionPlanner service has been called')
+
+def main(args=None):
+    """
+    Main function with example usage of the MissionPlannerClient.
+
+    """
+    rclpy.init(args=args)
+    mission_planner_client = MissionPlannerClient()
+    ox, oy = [], []
+    for i in range(-10, 60):
+        ox.append(i)
+        oy.append(60)
+    for i in range(20, 60):
+        ox.append(60)
+        oy.append(i)
+    for i in range(-10, 60):
+        ox.append(i)
+        oy.append(60)
+    for i in range(20, 61):
+        ox.append(-10)
+        oy.append(i)
+    for i in range(-10, 60):
+        ox.append(i)
+        oy.append(55)
+    for i in range(-10, 46):
+        ox.append(i)
+        oy.append(45)
+    for i in range(35, 45):
+        ox.append(45)
+        oy.append(i)
+    for i in range(-10, 46):
+        ox.append(i)
+        oy.append(35)
+    for i in range(-10, 60):
+        ox.append(i)
+        oy.append(25)
+    for i in range(-10, 60):
+        ox.append(i)
+        oy.append(20)
```

Comment:
Could you explain what is happening here? Is this creating the "closed-grid" for the D* algo? If so, this should maybe be moved to D* class module.

The idea in the end is to have a system which takes discrete and sparse points from the landmark server which are the obstacles. The client should not be responsible for creating the "fence" for the D* algo.

Reply:
Yeah, I agree that is messy. The idea was that that was an example of how the obstacle grid would look like after the actual obstacle coordinates from landmark went through a "grid creating function". This example tried to recreate a simplified version of the navigation task for Njord (https://njord.gitbook.io/team-handbook-njord-2024/9-task-descriptions/9.2-navigation), and should of course be replaced (Issue #140 ). However, since the priority was to have the path planner working, the complete logic of the mission planner is not yet complete and the mission parameters are just placeholders. Do let me know what you think about that.